website/orders/models.py

```diff
@@ -498,6 +567,8 @@ class Order(models.Model):
     paid = models.BooleanField(default=False, null=False)
     paid_at = models.DateTimeField(null=True, blank=True)
 
+    type = models.PositiveIntegerField(choices=TYPES, default=0, null=False, blank=False)
```

Comment:
This shouldn't be here as it can be derived from the product (check if the product is orderable/takeaway)

Reply:
Yes but a product can be orderable AND scannable. This is merely the way the product was registered (for example, scanned orders usually have no user associated with them).

Reply:
I see... We should document this, as it is pretty confusing

(You can now have orders:

- with and without a user attached
- scanned or 'ordered via the form' (I think it is not in order ordering ordered orders)
- products that can be ordered or not

this feels unnecessarily complex)

Reply:
Yes but don't you think it might be handy in the future to store the way an order was registered. We can also set the user field to the one who scanned the order if the type = Scanned. This will remove some of the options you pointed at.

Reply:
I agree